ReactWindows/ReactNative.Shared/Views/View/ReactViewManager.cs

```diff
@@ -92,6 +99,192 @@ protected override BorderedCanvas CreateViewInstance(ThemedReactContext reactCon
             return new BorderedCanvas();
         }
 
+        #region drag and drop
+#if WINDOWS_UWP
+
+        /// <summary>
+        /// Enables the Canvas as a drop target.
+        /// </summary>
+        [ReactProp("allowDrop")]
+        public void SetAllowDrop(BorderedCanvas view, bool allowDrop)
+        {
+            view.AllowDrop = allowDrop;
+
+            if (allowDrop)
+            {
+                view.DragEnter += OnDragEnter;
+                view.DragOver += OnDragOver;
+                view.Drop += OnDrop;
+                view.DragLeave += OnDragLeave;
+
+            }
+            else
+            {
+                view.DragEnter -= OnDragEnter;
+                view.DragOver -= OnDragOver;
+                view.Drop -= OnDrop;
+                view.DragLeave -= OnDragLeave;
+            }
+        }
+
+        private async Task<JObject> GetDataTransferInfo(DataPackageView data, bool drop = false)
+        {
+            var files = new JArray();
+            var items = new JArray();
+            var types = new JArray();
+
+            if (data.Contains(StandardDataFormats.StorageItems))
+            {
+                foreach (var item in await data.GetStorageItemsAsync())
```

Comment:
>GetStorageItemsAsync [](start = 48, length = 20)

We should double check if we want to support any other upload types like folders. See what the other clients do.

Reply:
Folders aren't supported in the UWP client.